iavl/iavl_test.go

```diff
@@ -570,6 +584,93 @@ func TestIAVLTreeProof(t *testing.T) {
 	}
 }
 
+func TestIAVLTreeKeyExistsProof(t *testing.T) {
```

Comment:
Do you want to move this to another test file (iavl_proof_test.go ?) I think it can be nice to separate old and new tests and code, like one file is refactored, the other needs to be....